panic: page 87 already freed

I've been doing a bit of corruption testing under unexpected power off scenarios (using an external USB drive and disconnecting it midstream) and, unfortunately, I'm fairly consistently hitting unrecoverable database corruption issues due to things such as `unreachable unfreed` or `already freed`.  I understand write corruption in this scenario is expected, but the database shouldn't be left in an unrecoverable state.

EDIT: I should also note this is on Windows 7 using all default bolt options.

I have a 4KiB database that is corrupted from this scenario that I'll keep around to help debug.  To get started, I'm providing the following pertinent details:
- A trace of the panic.  NOTE: The trace is from a vendored version of bolt, but there are no code modifications and it is synced through commit ee954308d64186f0fc9b7022b6178977848c17a3.
- Output of the `bolt pages` command
- Output of the `bolt check` command.  **NOTE: This command hangs and never completes**

```
goroutine 12 [running]:
github.com/btcsuite/bolt.(*freelist).free(0xc08202b680, 0xca2, 0x2af7000)
       .../bolt/freelist.go:118 +0x35c
github.com/btcsuite/bolt.(*node).spill(0xc0820e2380, 0x0, 0x0)
        .../bolt/node.go:349 +0x280
github.com/btcsuite/bolt.(*node).spill(0xc0820e2310, 0x0, 0x0)
        .../bolt/node.go:336 +0x134
github.com/btcsuite/bolt.(*node).spill(0xc0820e22a0, 0x0, 0x0)
        .../bolt/node.go:336 +0x134
github.com/btcsuite/bolt.(*Bucket).spill(0xc0820dc7c0, 0x0, 0x0)
        .../bolt/bucket.go:536 +0x1cb
github.com/btcsuite/bolt.(*Bucket).spill(0xc0820dc780, 0x0, 0x0)
        .../bolt/bucket.go:503 +0xac9
github.com/btcsuite/bolt.(*Bucket).spill(0xc0820e65c8, 0x0, 0x0)
        .../bolt/bucket.go:503 +0xac9
github.com/btcsuite/bolt.(*Tx).Commit(0xc0820e65b0, 0x0, 0x0)
        .../bolt/tx.go:150 +0x1f5
...
```

```bash
$ bolt pages test.db

ID       TYPE       ITEMS  OVRFLW
======== ========== ====== ======
0        meta       0            
1        meta       0            
2        leaf       14           
3        leaf       19           
4        leaf       19           
5        leaf       17           
6        leaf       17           
7        leaf       27           
8        leaf       22           
9        leaf       17           
10       leaf       17           
11       leaf       15           
12       leaf       19           
13       leaf       26           
14       leaf       14           
15       leaf       17           
16       leaf       25           
17       leaf       17           
18       leaf       16           
19       leaf       22           
20       leaf       17           
21       leaf       15           
22       leaf       27           
23       leaf       22           
24       leaf       16           
25       leaf       16           
26       leaf       24           
27       leaf       24           
28       leaf       27           
29       leaf       23           
30       leaf       23           
31       leaf       18           
32       leaf       26           
33       leaf       16           
34       leaf       22           
35       leaf       16           
36       leaf       25           
37       leaf       19           
38       leaf       23           
39       leaf       14           
40       leaf       25           
41       leaf       28           
42       leaf       16           
43       leaf       16           
44       leaf       16           
45       leaf       20           
46       leaf       18           
47       leaf       26           
48       leaf       18           
49       leaf       24           
50       leaf       28           
51       leaf       23           
52       leaf       21           
53       leaf       18           
54       leaf       24           
55       leaf       20           
56       leaf       27           
57       leaf       18           
58       leaf       18           
59       leaf       23           
60       leaf       28           
61       leaf       26           
62       leaf       17           
63       leaf       16           
64       leaf       26           
65       leaf       19           
66       leaf       15           
67       leaf       14           
68       leaf       17           
69       leaf       18           
70       leaf       27           
71       leaf       21           
72       leaf       23           
73       leaf       16           
74       leaf       19           
75       leaf       23           
76       leaf       19           
77       leaf       21           
78       leaf       17           
79       leaf       24           
80       leaf       25           
81       leaf       16           
82       leaf       27           
83       leaf       15           
84       leaf       21           
85       leaf       22           
86       leaf       23           
87       freelist   6            
88       leaf       16           
89       leaf       15           
90       leaf       22           
91       leaf       19           
92       leaf       14           
93       leaf       18           
94       leaf       19           
95       leaf       18           
96       leaf       21           
97       leaf       23           
98       leaf       19           
99       leaf       25           
100      leaf       19           
101      leaf       17           
102      leaf       22           
103      leaf       16           
104      leaf       25           
105      leaf       20           
106      leaf       17           
107      leaf       22           
108      leaf       15           
109      leaf       16           
110      leaf       21           
111      leaf       15           
112      leaf       14           
113      leaf       27           
114      leaf       17           
115      leaf       15           
116      leaf       21           
117      leaf       26           
118      leaf       26           
119      leaf       22           
120      leaf       20           
121      leaf       16           
122      leaf       25           
123      leaf       15           
124      leaf       18           
125      leaf       18           
126      leaf       16           
127      leaf       28           
128      leaf       18           
129      leaf       17           
130      leaf       19           
131      leaf       25           
132      leaf       26           
133      leaf       24           
134      leaf       18           
135      leaf       26           
136      leaf       20           
137      leaf       18           
138      leaf       23           
139      leaf       26           
140      leaf       15           
141      leaf       17           
142      leaf       24           
143      leaf       16           
144      leaf       16           
145      leaf       22           
146      leaf       15           
147      leaf       18           
148      leaf       26           
149      leaf       14           
150      leaf       14           
151      leaf       14           
152      leaf       15           
153      leaf       15           
154      leaf       17           
155      leaf       27           
156      leaf       22           
157      branch     80           
158      leaf       15           
159      leaf       15           
160      leaf       15           
161      leaf       15           
162      leaf       15           
163      leaf       15           
164      branch     42           
165      branch     45           
166      free                    
167      branch     4            
168      free                    
169      leaf       2            
170      leaf       1            
171      free                    
172      free                    
173      free                    
174      free                    
```

```
$ bolt check test.db
page 87: reachable freed
page 105: multiple references
page 83: multiple references
page 87: multiple references
page 87: reachable freed
page 97: multiple references
page 97: multiple references
page 97: multiple references
page 157: multiple references
page 21: multiple references
page 20: multiple references
page 87: multiple references
page 87: reachable freed
page 27: multiple references
page 35: multiple references
page 129: multiple references
page 132: multiple references
page 156: multiple references
page 82: multiple references
page 80: multiple references
page 139: multiple references
page 60: multiple references
page 121: multiple references
page 153: multiple references
page 30: multiple references
page 7: multiple references
page 112: multiple references
page 9: multiple references
page 127: multiple references
page 118: multiple references
page 74: multiple references
page 106: multiple references
page 15: multiple references
page 78: multiple references
page 65: multiple references
page 39: multiple references
page 159: multiple references
page 63: multiple references
page 124: multiple references
page 107: multiple references
page 3: multiple references
page 16: multiple references
page 67: multiple references
page 163: multiple references
page 29: multiple references
page 92: multiple references
page 105: multiple references
page 105: multiple references
page 13: multiple references
page 85: multiple references
page 109: multiple references
page 137: multiple references
page 48: multiple references
page 138: multiple references
page 75: multiple references
page 28: multiple references
page 83: multiple references
page 55: multiple references
page 32: multiple references
page 34: multiple references
page 84: multiple references
page 51: multiple references
page 53: multiple references
page 126: multiple references
page 19: multiple references
page 23: multiple references
page 95: multiple references
page 96: multiple references
page 5: multiple references
page 37: multiple references
page 130: multiple references
page 149: multiple references
page 161: multiple references
page 72: multiple references
page 73: multiple references
page 116: multiple references
page 113: multiple references
page 18: multiple references
page 125: multiple references
page 146: multiple references
page 102: multiple references
page 134: multiple references
page 76: multiple references
page 114: multiple references
page 62: multiple references
page 66: multiple references
page 103: multiple references
page 97: multiple references
page 83: multiple references
page 87: multiple references
page 87: reachable freed
**Hangs here and never completes** 
```

```
panic: page 4 already freed

goroutine 818021 [running]:
github.com/boltdb/bolt.(*freelist).free(0xc2fd236c90, 0x5b0, 0x7ecaaacc4000)
        /source/.gopack/vendor/src/github.com/boltdb/bolt/freelist.go:117 +0x355
github.com/boltdb/bolt.(*node).spill(0xc48b570e70, 0x0, 0x0)
        /source/.gopack/vendor/src/github.com/boltdb/bolt/node.go:358 +0x279
github.com/boltdb/bolt.(*Bucket).spill(0xc55bcffdc0, 0x0, 0x0)
        /source/.gopack/vendor/src/github.com/boltdb/bolt/bucket.go:536 +0x1c4
github.com/boltdb/bolt.(*Bucket).spill(0xc55bcffd80, 0x0, 0x0)
        /source/.gopack/vendor/src/github.com/boltdb/bolt/bucket.go:503 +0xac2
github.com/boltdb/bolt.(*Bucket).spill(0xc6581b6b78, 0x0, 0x0)
        /source/.gopack/vendor/src/github.com/boltdb/bolt/bucket.go:503 +0xac2
github.com/boltdb/bolt.(*Tx).Commit(0xc6581b6b60, 0x0, 0x0)
        /source/.gopack/vendor/src/github.com/boltdb/bolt/tx.go:151 +0x1ee
github.com/boltdb/bolt.(*DB).Update(0xc27dc99380, 0xca1e051d68, 0x0, 0x0)
        /source/.gopack/vendor/src/github.com/boltdb/bolt/db.go:554 +0x169
customerio/index/index.(*DBMBolt).UpdateShard(0xc20800d5f0, 0x90da, 0xc5a9d20a50, 0x0, 0x0)
        /source/src/customerio/index/index/db_mbolt.go:448 +0xc9
customerio/index/index.func·057(0xc5000090da)
        /source/src/customerio/index/index/db_memory.go:359 +0x503
created by customerio/index/index.(*DBMemory).Flush
        /source/src/customerio/index/index/db_memory.go:367 +0x34a
```

--- a/node.go
+++ b/node.go
@@ -463,43 +463,6 @@ func (n *node) rebalance() {
 		target = n.prevSibling()
 	}
 
-	// If target node has extra nodes then just move one over.
-	if target.numChildren() > target.minKeys() {
-		if useNextSibling {
-			// Reparent and move node.
-			if child, ok := n.bucket.nodes[target.inodes[0].pgid]; ok {
-				child.parent.removeChild(child)
-				child.parent = n
-				child.parent.children = append(child.parent.children, child)
-			}
-			n.inodes = append(n.inodes, target.inodes[0])
-			target.inodes = target.inodes[1:]
-
-			// Update target key on parent.
-			target.parent.put(target.key, target.inodes[0].key, nil, target.pgid, 0)
-			target.key = target.inodes[0].key
-			_assert(len(target.key) > 0, "rebalance(1): zero-length node key")
-		} else {
-			// Reparent and move node.
-			if child, ok := n.bucket.nodes[target.inodes[len(target.inodes)-1].pgid]; ok {
-				child.parent.removeChild(child)
-				child.parent = n
-				child.parent.children = append(child.parent.children, child)
-			}
-			n.inodes = append(n.inodes, inode{})
-			copy(n.inodes[1:], n.inodes)
-			n.inodes[0] = target.inodes[len(target.inodes)-1]
-			target.inodes = target.inodes[:len(target.inodes)-1]
-		}
-
-		// Update parent key for node.
-		n.parent.put(n.key, n.inodes[0].key, nil, n.pgid, 0)
-		n.key = n.inodes[0].key
-		_assert(len(n.key) > 0, "rebalance(2): zero-length node key")
-
-		return
-	}
-
 	// If both this node and the target node are too small then merge them.
 	if useNextSibling {
 		// Reparent all child nodes being moved.

